usl_pipeline/usl_lib/usl_lib/readers/hdf5_readers.py

```diff
@@ -0,0 +1,29 @@
+import typing
+from typing import Dict
+
+import h5py
+import numpy
+import numpy.typing as npt
+
+
+def read_from_hdf5(
+    input_stream: typing.BinaryIO,
+    group_name: str = "features",
+) -> Dict[str, npt.NDArray[numpy.float32]]:
+    """Reads feature matrix layers from HDF5 file.
+
+    Args:
+        input_stream: Input byte stream to read from.
+        group_name: Name of HDF5 group where layers are stored.
+
+    Returns:
+        Dictionary with keys corresponding to feature layer names and values containing
+        Numpy 2d arrays of each feature layer.
+    """
+    hf = h5py.File(input_stream, "r")
```

Comment:
```python
with h5py.File(input_stream, "r") as hf
```
would prevent us from having to `hf.close()` ourselves.

Reply:
Done